core/src/main/scala/kafka/network/SocketServer.scala

```diff
@@ -173,21 +194,30 @@ class SocketServer(val config: KafkaConfig, val metrics: Metrics, val time: Time
     info("Stopping socket server request processors")
     this.synchronized {
       acceptors.asScala.values.foreach(_.shutdown())
-      processors.asScala.values.foreach(_.shutdown())
-      requestChannel.clear()
+      dataProcessors.asScala.values.foreach(_.shutdown())
+      controlPlaneProcessors.asScala.values.foreach(_.shutdown())
+      dataRequestChannel.clear()
+      if (controlPlaneRequestChannel != null)
+        controlPlaneRequestChannel.clear()
       stoppedProcessingRequests = true
     }
     info("Stopped socket server request processors")
   }
 
   def resizeThreadPool(oldNumNetworkThreads: Int, newNumNetworkThreads: Int): Unit = synchronized {
     info(s"Resizing network thread pool size for each listener from $oldNumNetworkThreads to $newNumNetworkThreads")
+    val dataAcceptors = config.controlPlaneListenerName match {
+      case Some(controlPlaneListenerName) =>
+        acceptors.asScala.filter{ case (endpoint, _) => !endpoint.listenerName.value().equals(controlPlaneListenerName.value())}
+      case None => acceptors.asScala
+    }
+
     if (newNumNetworkThreads > oldNumNetworkThreads) {
-      acceptors.asScala.foreach { case (endpoint, acceptor) =>
-        addProcessors(acceptor, endpoint, newNumNetworkThreads - oldNumNetworkThreads)
+      dataAcceptors.foreach { case (endpoint, acceptor) =>
+        addProcessors(acceptor, endpoint, newNumNetworkThreads - oldNumNetworkThreads, dataRequestChannel, dataProcessors, false)
```

Comment:
With the above suggestion, this would be : 
```suggestion
        addDataProcessors(acceptor, endpoint, newNumNetworkThreads - oldNumNetworkThreads)
```